src/Rox/Core/Core.php

```diff
@@ -229,11 +231,13 @@ public function setup(
             if (!preg_match($mongoApiKeyPattern, $sdkSettings->getApiKey()) && !preg_match($uuidApiKeyPattern, $sdkSettings->getApiKey())) {
                 throw new InvalidArgumentException("Illegal rollout apikey");
             }
+
+            if (preg_match($uuidApiKeyPattern, $sdkSettings->getApiKey())) {
+                $this->_environment->setToPlatform();
```

Comment:
this will override the roxOptions.networkConfigurations

if we want to use preprod, it will be a uuid key
but it looks from the code here, that if it's a uuid it will use production values, regardless
we need to be able to connect to preprod, probably the priorities should be:
1. roxOptions.networkconfigs
2. roxOptions.apiKey is uuid